Initial review feedback

This is the feedback from my initial review of the workshop:

- [x] #52 (marking as completed since it's a consideration that went into a separate ticket)
- [x] slide 6 typo: "a remote is AN module" -> "a remote is A module" https://github.com/nearform/the-micro-frontends-workshop/pull/45
- [x] why is yarn install (which btw can also be shortened to simply yarn) required in each step? are we not using workspaces?
> I think it's more just to make sure the dependencies are installed in case a user skips some steps
- [x] slide 11: "steps that need to be made", I believe a better form is that steps are taken, not made
- [x] slide 12: the code shown is different from the one in the step, specifically the "name" property of the module federation plugin constructor options
- [x] slide 14 typo: "to be app's entry point" -> "to be THE app's entry point"
- [x] slide 21: it says that in the first link you should see a page that displays a button, but I see a page that displays a list of links (this is also confirmed by the following slide, so I assume slide 21 is outdated) https://github.com/nearform/the-micro-frontends-workshop/pull/64
- [x] #54
  - it's really not clear whether I should implement the code or if the code is already there, but then you realize that the code is already there. So what's a user supposed to do? I mean if we're expecting the user to follow along and write code, then we should provide a playground that starts out from the basic app and where the user works. We have done this in other workshops. The steps should already have the solutions so the user can copy/inspire, but without a playground the user is basically prevented from writing any code, as all the code is already written for them
  - not clear how the app works. I mean I can understand that this is an app that exposes a remote but which also hosts a whole app, although it doesn't consume any remote components. I think it'd be worth clarifying it in the exercise

- [x] slide 25: why are we using next here? isn't it an unnecessary complication? why are we not using a plain react app?
- [x] #58 
- [x] slide 26: instructions unclear. it doesn't say where and how to import the Nav remote component nor where to render the LayoutBox component. The first of which is especially important because never in the slides was shown how to actually consume a remote component. #58
- [x] slide 27: why is !isServer needed? Again I think we should do without Next, it doesn't seem to bring anything useful to the picture
- [x] yarn start in step 2 doesn't work on my machine: https://github.com/nearform/the-micro-frontends-workshop/pull/55

```
simone@X86 MINGW64 /n/nearform/the-micro-frontends-workshop/src/step-02-setting-up-host (master)
$ yarn start
yarn run v1.22.19
$ concurrently "cd ../step-01-setting-up-remote; npm start" "npm run dev"
[0] The system cannot find the path specified.
[0] cd ../step-01-setting-up-remote; npm start exited with code 1
[1]
[1] > step-02-setting-up-host@1.0.0 dev
[1] > next dev -p 8080
[1]
[1] - ready started server on 0.0.0.0:8080, url: http://localhost:8080
[1] No ModuleFederationPlugin(s) found.
[1] (node:5176) [DEP_WEBPACK_MODULE_ID] DeprecationWarning: Module.id: Use new ChunkGraph API
[1] (Use `node --trace-deprecation ...` to show where the warning was created)
[1] - event compiled client and server successfully in 614 ms (54 modules)
[1] - wait compiling...
[1] - event compiled client and server successfully in 308 ms (54 modules)
```

- [x] step 03: why is the start command using concurrently to run the apps in the workspaces? that's one reason why we use workspaces, the package manager can run commands in the workspaces without us having to use fancy tools https://github.com/nearform/the-micro-frontends-workshop/pull/55
- [x] the start command for step 3 doesn't work on my machine (windows) https://github.com/nearform/the-micro-frontends-workshop/pull/55

```
simone@X86 MINGW64 /n/nearform/the-micro-frontends-workshop/src/step-03-bi-directional (master)
$ npm start

> bi-directional@1.0.0 start
> concurrently "cd reactApp; npm start" "cd nextApp; npm run dev" --kill-others

[0] The system cannot find the path specified.
[0] cd reactApp; npm start exited with code 1
--> Sending SIGTERM to other processes..
[1] The system cannot find the path specified.
[1] cd nextApp; npm run dev exited with code 1
npm ERR! Lifecycle script `start` failed with error:
npm ERR! Error: command failed
npm ERR!   in workspace: bi-directional@1.0.0
npm ERR!   at location: N:\nearform\the-micro-frontends-workshop\src\step-03-bi-directional
```

- [x] slide 51 cut at the bottom
![image](https://github.com/nearform/the-micro-frontends-workshop/assets/20181/ec87cad3-23eb-4742-a47e-f8ae3a3ef3f3) https://github.com/nearform/the-micro-frontends-workshop/pull/55
- [x] slide 52: use inline links instead of long form, for readability https://github.com/nearform/the-micro-frontends-workshop/pull/55
- [x] slide 52: why is the workshop github link in the references? https://github.com/nearform/the-micro-frontends-workshop/pull/55

--- a/slides.md
+++ b/slides.md
@@ -31,7 +31,9 @@ The concept of <em>micro frontends</em> was first mentioned circa 2016 as an ext
 
 <div class="bigger">
 
-Module federation is one of the most popular approaches for implementing micro frontend architecture on either the client or server side. With module federation, each micro frontend is treated as a standalone module that can be developed, deployed, and versioned independently, which allows those modules to share and consume each other's functionality, resources, and components, improving collaboration and reusability. Started as a Webpack Plugin, module federation has now evolved into a general concept adopted by other bundlers and frameworks.
+Module federation is one of the most popular approaches for implementing micro frontend architecture on either the client or server side. With module federation, each micro frontend is treated as a standalone module that can be developed, deployed, and versioned independently, which allows those modules to share and consume each other's functionality, resources, and components at runtime, improving collaboration and reusability.
+
+Started as a Webpack Plugin, module federation has now evolved into a general concept adopted by other bundlers and frameworks.
 
 </div>
 
@@ -70,11 +72,11 @@ Module federation is one of the most popular approaches for implementing micro f
 
 <div class="dense">
 
--- a **host** is an application that includes the initial chunks of our code, the ones that will be used to bootstrap our container – as part of module federation, those are just referenced to a remote, which allows for smaller bundle sizes and shorter initial load times;
+-- a **host** is an application that includes the initial chunks of our code, the ones that will be used to bootstrap our container; the concept of module federation assumes that some components that container will render are just being referenced to a remote and not the initial bundle, which allows for smaller bundle sizes and shorter initial load times;
 
 -- a **remote** is a module that is being consumed by the host, and it can be either shared components or common dependencies to be used by different hosts;
 
--- a **bidirectional host** can be either a host or a remote (or both), consuming other applications or providing some code to other hosts.
+-- a **bidirectional host** is both a host and a remote, consuming other remotes and providing some code to other hosts.
 </div>
 
 ---
@@ -85,7 +87,7 @@ Module federation is one of the most popular approaches for implementing micro f
 <p>As mentioned earlier, initially, module federation was implemented as a plugin introduced in Webpack 5.
 To set up Module Federation in Webpack, you need to define the federated modules in your Webpack configuration files, specify the remote entry points and expose specific modules (aka “remotes”) that you want to share with other applications. The remote entry points represent the Webpack builds that expose modules for consumption.</p>
 
-<p>In the consuming application's (aka “host”) Webpack configuration, you define which federated modules you want to consume. You specify the remote entrypoints and the modules you want to import from those remotes.
+<p>In the consuming application's (aka “host”) Webpack configuration, you define which federated modules you want to consume. You specify the remote entry points and the modules you want to import from those remotes.
 When you build and run your applications, Webpack dynamically loads the federated modules at runtime. It fetches the remote entry points, resolves the requested modules, and injects them into the consuming application. This process allows you to share code between applications without physically bundling everything together.</p>
 
 <p>Note that as an application has multiple dependencies, a host can also have multiple remotes.</p>
@@ -223,7 +225,7 @@ plugins: [
 
 ### Adding an extra layer of indirection to the entire app
 
-We need `index.js` to be the app's entry point but inside of it we need to import another file, `bootstrap.js` (named this way by convention) that renders the entire app. This file contains what `index.js` would normally contain in a React app including `createRoot()` method. To allow Module Federation we need to import it dynamically using `import()` inside of `index.js`.
+We need `index.js` to be the app's entry point but inside of it we need to import another file, `bootstrap.js` (named this way by convention) that renders the entire app. This file contains what `index.js` would normally contain in a React app including `createRoot()` method. To allow Module Federation we need to import it dynamically using `import()` inside of `index.js`. This extra layer is needed for Webpack to load all of the imports necessary to render the remote app (failure to provide it will result in an error saying something like `Shared module is not available for eager consumption`).
 
 ```js
 //src/bootstrap.js
@@ -235,12 +237,6 @@ root.render(<App />);
 ```js
 //src/index.js
 import('bootstrap.js')
-```
-
-Without this extra layer of indirection Webpack would throw the following error:
-
-```js
-Shared module is not available for eager consumption
 ```
 
 </div>
@@ -370,9 +366,9 @@ http://localhost:3002/remoteEntry.js
 
 ```
 
-For the first one, you should see a page that displays the `Basic Remote Application` title with a `Remote Button`.
+For the first one, you should see a page that displays the `Basic Remote Application` title with your `Nav` component (a list of links).
 
-For the second one, you should see a `script` that exposes our button element for remote consumption.
+For the second one, you should see a `script` that exposes our component for remote consumption.
 
 </div>
 
@@ -397,7 +393,7 @@ For the second one, you should see a `script` that exposes our button element fo
 
 -- In this step we are going to set up an application as the host application. We are going to use a Next.js application for this purpose which will demonstrate Module Federation's ability to work with different frameworks.
 
-### General Steps: 
+### General Steps:
 
 -- In order to start consuming modules, we need to configure the plugin's `remotes` parameter which can take multiple remotes.
 
@@ -440,7 +436,7 @@ const SomeRemoteComponent = dynamic(() => import('someRemoteApp/SomeRemoteCompon
 
 </div>
 
---- 
+---
 
 ## Step 2: Setting up the Host Application /3
 
@@ -498,7 +494,7 @@ config.plugins.push(
 // next.config.js
 const NextFederationPlugin = require('@module-federation/nextjs-mf');
 module.exports = {
-  webpack(config) {    
+  webpack(config) {
       config.plugins.push(
         new NextFederationPlugin({
           name: 'nextApp',
@@ -536,8 +532,8 @@ export default function Home() {
       <Head>
         {/* Head example code */}
       </Head>
-      <LayoutBox>              
-        <Nav links={links} />       
+      <LayoutBox>
+        <Nav links={links} />
       </LayoutBox>
     </div>
   )
@@ -553,7 +549,7 @@ export default function Home() {
 <div class="dense">
 
 
-In your terminal from the `src/step-02-setting-up-host` folder run: 
+In your terminal from the `src/step-02-setting-up-host` folder run:
 
 ```bash
 yarn && yarn start
@@ -826,7 +822,7 @@ Please follow the guidelines from Zack Jackson (inventor & co-creator of module 
 
 -- **singleton (boolean)**: whether the dependency will be considered a singleton, which means that only a single instance of it is supposed to be shared across all the federated modules;
 
--- **requiredVersion (string)**: specifies the required version of the dependency, which makes any incompatible version loaded separately (not shared); note that if the `singleton` property setting `requiredVersion` will raise a warning in case of a conflict.
+-- **requiredVersion (string)**: specifies the required version of the dependency, which makes any incompatible version loaded separately (not shared); note that if the `singleton` property is set to `true`, setting `requiredVersion` will raise a warning in case of a conflict.
 
 ---
 
@@ -965,9 +961,9 @@ const packageJsonDependencies = require('./package.json').dependencies
 <p>
 When it comes to TypeScript applications, the most common problem with using external libraries (which can be federated remote modules) is that not all of them provide TypeScript types with the original code. In the context of module federation, this problem is aggravated by the fact that Webpack only loads resources from the federated module at runtime, TypeScript, however, needs those during compilation. In this case you have the following options:</p>
 
--- @module-federation/typescript,
+-- @module-federation/native-federation-typescript,
 
--- @module-federation/native-federation-typescript.
+-- @module-federation/typescript,
 
 -- packaging your types for distribution via a package registry (e.g., npm);
 

